import_export/admin.py

```diff
@@ -119,7 +119,11 @@ def process_import(self, request, *args, **kwargs):
     def process_dataset(self, dataset, confirm_form, request, *args, **kwargs):
 
         res_kwargs = self.get_import_resource_kwargs(request, form=confirm_form, *args, **kwargs)
-        resource = self.get_import_resource_class()(**res_kwargs)
+        if request.POST.get('resource', None):
+            resource_class = self.get_import_resource_class()[int(request.POST['resource'])]
```

Comment:
One option we could consider is that we change this field to `resource_classes` - then it is explicit that this is expected to be a list (which could be length 1).

This would be a breaking change but it could go into our v3 release with other breaking changes.  What do you think?

Reply:
I really don't know here. I didn't want to break things and require implementators to change their code only to be semantically exact.

We could also use `resource_classes` only if there are multiple resources. Or we can accept both `resource_class` and `resource_classes` if one of them is set.

Reply:
Hi, I'd say that we should be friendly and keep `resource_class` adding a deprecation warning, then add a `get_resource_classes` method that will look both into `resource_class` and `resource_classes`. And maybe warn that both can't be set at the same time. What do you think?

Reply:
I agree with both of you.  This is a tricky issue.  On one hand it is a minor issue because it is solely semantic, but on the other hand it seems wrong to have an attribute named `resource_class` which could be a list of classes.

I agree that we should be friendly.  Supporting both would not be too tricky.  Initially only users who want multiple classes will declare `resource_classes`.

I have been thinking about whether it makes sense to keep both, but I think that would cause confusion in the long run.

I think that supporting both, and adding a deprecation warning to `resource_class` is the way to go.

Reply:
And should the `resource_classes` support only list/tuple (simillar to `display_names`) and not the single class?
It would be still semantically wrong to have `resource_classes = Resource`.

Reply:
I also think that v3 will involve breaking changes anyway, so users will have to change their implementations, so maybe that is the opportunity to move to `resource_classes`.  Users who couldn't do this yet could stay on the 2.x releases.

Reply:
> And should the resource_classes support only list/tuple (simillar to display_names) and not the single class?

Yes I think it should only support list/tuple

Reply:
OK. I will make implement only the `resource_classes` with exception thrown if any of `resource_class` or the `get_*_resource_class` is implemented. I think we should not let users to run it if they didn't migrated their code, because it would fall back to the `modelresource_factory` silently, which could cause problems.

Reply:
There's some good general guidance [here](https://www.python.org/dev/peps/pep-0387/#making-incompatible-changes)

Reply:
How easy would it be to make it backwards compatible with using `resource_class`?  Could we log a deprecation warning rather than raising exceptions (some guidance [here](https://www.lesinskis.com/python_deprecation_tutorial.html)).

I think the ideal solution for now is that you can use either `resource_class` or `resource_classes`, but the former is deprecated and will be removed in a future release.

Sorry to be a pain - I know we are going back and forth on this - I think it is nearly there and it is worth getting it right.

Reply:
OK. I managed to add that.
Although the deprecation warings will show only with the `-Wd` python option.

Reply:
This is fine and consistent with Django's approach.  I see the warnings in my logs as well.

Reply:
btw - please can you update the docs you added as well (users can no longer assign a list to `resource_class`).

Reply:
I updated the docs and set the new default `resource_classes` for the examples.

BTW. There is no mention of the `get_resource_classes()` (and import/export variants) in the [autogenerated admin docs](https://django-import-export.readthedocs.io/en/latest/api_admin.html).